File: baml_language/sdk_tests/crates/swift/type_shapes/customizable/TestMain.swift
// Namespace/type reachability — port of python_pydantic2's type_shapes
// `test_main.py` (its pytest half; the pyright half maps to `swift
// build` itself, which type-checks the whole generated package).
//
// Python touches every generated namespace module; Swift touches each
// namespace enum's metatype, which fails to compile if the symbol
// vanishes. Not yet reachable (no supported symbols emitted, so no
// namespace enum exists): `media` (media types, Phase 5).
import XCTest
import Baml

final class TestMain: XCTestCase {
    func test_main_all_namespaces_reachable() {
        _ = Baml.primitives.self
        _ = Baml.enums.self
        _ = Baml.literals.self
        _ = Baml.class_refs.self
        _ = Baml.aliases.self
        _ = Baml.aliases_consumer.self
        _ = Baml.optional.self
        _ = Baml.lists.self
        _ = Baml.maps.self
        _ = Baml.unions.self
        _ = Baml.recursion.self
        _ = Baml.generics.self
        _ = Baml.forward_refs.self
        _ = Baml.complex_models.self
        _ = Baml.lorem.self
        _ = Baml.a.self
    }

    func test_main_root_foo_reachable() throws {
        let f = try Baml.round_trip_foo(f: Baml.Foo(v: 3))
        XCTAssertEqual(f, Baml.Foo(v: 3))
    }

    func test_main_lorem_resume_reachable() {
        _ = Baml.lorem.Resume.self
    }

    func test_main_deep_namespace_thing_reachable() {
        _ = Baml.a.b.Thing.self
    }
}

File: baml_language/sdk_tests/crates/java/type_shapes/customizable/TestMain.java
// Smoke tests for the type_shapes sdk-test crate.
//
// Port of python_pydantic2/type_shapes/customizable/test_main.py — same
// test names, same intent.
//
// java-port note: Python's smoke tests are `import baml_sdk.<ns>` module
// imports — a namespace "imports cleanly" if the import statement doesn't
// raise. Per the conventions doc's "Namespace-import smoke tests" rule,
// the Java analog is referencing a known generated symbol's `.class`
// (compile-time reachability, plus the class-load side effect that
// triggers runtime init per the doc's "Runtime init" row). Each namespace
// below is represented by one real symbol declared in that namespace's
// fixture BAML source.
import static org.junit.jupiter.api.Assertions.assertNotNull;

import baml_sdk.Foo;
import baml_sdk.a.b.Thing;
import baml_sdk.lorem.Resume;
import org.junit.jupiter.api.Test;

class TestMain {

    @Test
    void test_main_root_imports_cleanly() {
        assertNotNull(baml_sdk.Fns.class);
    }

    @Test
    void test_main_all_namespaces_reachable() {
        assertNotNull(baml_sdk.primitives.Primitives.class);
        assertNotNull(baml_sdk.media.Media.class);
        assertNotNull(baml_sdk.enums.Sentiment.class);
        assertNotNull(baml_sdk.literals.Literals.class);
        assertNotNull(baml_sdk.class_refs.Outer.class);
        assertNotNull(baml_sdk.aliases.AliasContainer.class);
        assertNotNull(baml_sdk.aliases_consumer.RecListConsumer.class);
        assertNotNull(baml_sdk.optional.OptionalContainer.class);
        assertNotNull(baml_sdk.lists.ListContainer.class);
        assertNotNull(baml_sdk.maps.MapContainer.class);
        assertNotNull(baml_sdk.unions.UnionContainer.class);
        assertNotNull(baml_sdk.recursion.IntBinaryTree.class);
        assertNotNull(baml_sdk.generics.Wrapper.class);
        assertNotNull(baml_sdk.forward_refs.Other.class);
        assertNotNull(baml_sdk.complex_models.ComplexProfile.class);
        assertNotNull(baml_sdk.lorem.Resume.class);
        assertNotNull(baml_sdk.a.Fns.class);
    }

    @Test
    void test_main_root_foo_reachable() {
        assertNotNull(Foo.class);
    }

    @Test
    void test_main_lorem_resume_reachable() {
        assertNotNull(Resume.class);
    }

    @Test
    void test_main_deep_namespace_thing_reachable() {
        assertNotNull(Thing.class);
    }
}
